src/Profiling.cpp

```diff
@@ -538,77 +592,84 @@ class InjectProfiling : public IRMutator {
     s = Block::make({Evaluate::make(init_sampling_token), s});
-    s = LetStmt::make("profiler_shared_sampling_token",
+    s = LetStmt::make(names.profiler_shared_sampling_token,
                       Call::make(Handle(), Call::alloca, {Int(32).bytes()}, Call::Intrinsic), s);
 
-    s = LetStmt::make("profiler_pipeline_state", get_pipeline_state, s);
-    s = LetStmt::make("profiler_state", get_state, s);
     // If there was a problem starting the profiler, it will call an
     // appropriate halide error function and then return the
     // (negative) error code as the token.
-    s = Block::make(AssertStmt::make(profiler_token >= 0, profiler_token), s);
-    s = LetStmt::make("profiler_token", start_profiler, s);
+    s = Block::make(AssertStmt::make(profiler_start_error_code == 0, profiler_start_error_code), s);
+    s = LetStmt::make(names.profiler_start_error_code, start_profiler, s);
 
     if (!no_stack_alloc) {
         for (int i = num_funcs - 1; i >= 0; --i) {
-            s = Block::make(Store::make("profiling_func_stack_peak_buf",
+            s = Block::make(Store::make(names.profiler_func_stack_peak_buf,
                                         make_const(UInt(64), profiling.func_stack_peak[i]),
                                         i, Parameter(), const_true(), ModulusRemainder()),
                             s);
         }
-        s = Block::make(s, Free::make("profiling_func_stack_peak_buf"));
-        s = Allocate::make("profiling_func_stack_peak_buf", UInt(64),
+        s = Block::make(s, Free::make(names.profiler_func_stack_peak_buf));
+        s = Allocate::make(names.profiler_func_stack_peak_buf, UInt(64),
                            MemoryType::Auto, {num_funcs}, const_true(), s);
     }
 
     for (const auto &p : profiling.indices) {
-        s = Block::make(Store::make("profiling_func_names", p.first, p.second, Parameter(), const_true(), ModulusRemainder()), s);
+        s = Block::make(Store::make(names.profiler_func_names, p.first, p.second, Parameter(), const_true(), ModulusRemainder()), s);
     }
 
-    s = Block::make(s, Free::make("profiling_func_names"));
-    s = Allocate::make("profiling_func_names", Handle(),
+    s = Block::make(s, Free::make(names.profiler_func_names));
+    s = Allocate::make(names.profiler_func_names, Handle(),
                        MemoryType::Auto, {num_funcs}, const_true(), s);
     s = Block::make(Evaluate::make(stop_profiler), s);
 
+    // Allocate memory for the profiler instance state
+
+    // Check there isn't going to be end-of-struct padding to worry about.
+    static_assert((sizeof(halide_profiler_func_stats) & 7) == 0);
+
+    const int instance_size_bytes = sizeof(halide_profiler_instance_state) + num_funcs * sizeof(halide_profiler_func_stats);
+
+    s = Allocate::make(names.profiler_instance, UInt(64), MemoryType::Auto,
+                       {instance_size_bytes / 8}, const_true(), s);
```

Comment:
I guess you are missing the same static_assert check to test for the sizeof(halide_profiler_instance_state). If you omit this check, then the `(size + 7) / 8` trick should be applied.

Reply:
Will fix in next push.